scrabble.js

```diff
@@ -1,8 +1,157 @@
-var Scrabble = function() {};
+var Scrabble = function() {
+  // create the hashmap
+  this.letterScore = { "a": 1, "b": 3, "c": 3, "d": 2, "e": 1, "f": 4, "g": 2,
+  "h": 4, "i": 1, "j": 8, "k": 5, "l": 1, "m": 3, "n": 1, "o": 1, "p": 3,
+  "q": 10, "r": 1, "s": 1, "t": 1, "u": 1, "v": 4, "w": 4, "x": 8, "y": 4,
+  "z": 10 };
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+  this.maxLength = 7;
+  this.bonus = 50;
+  this.tileBag = new TileBag();
+};
+
+var TileBag = function() {
+  this.defaultTiles = ["a", "a", "a", "a", "a", "a", "a", "a",
+        "a", "n", "n", "n", "n", "n", "n", "b", "b", "o", "o", "o", "o", "o",
+        "o", "o", "o", "c", "c", "p", "p", "d", "d", "d", "d", "q", "e", "e",
+        "e", "e", "e", "e", "e", "e", "e", "e", "e", "e", "r", "r", "r", "r",
+        "r", "r", "f", "f", "s", "s", "s", "s", "g", "g", "g", "t", "t", "t",
+        "t", "t", "t", "h", "h", "u", "u", "u", "u", "i", "i", "i", "i", "i",
+        "i", "i", "i", "i", "v", "v", "j", "w", "w", "k", "x", "l", "l", "l",
+        "l", "y", "y", "m", "m", "z"];
+};
+
+TileBag.prototype.drawTiles = function(numOfTiles) {
+  this.randomTiles = [];
+  for(var i=0; i<numOfTiles; i++) {
+  // tiles are formatted like: [number, original index]
+    var tile = sample(this.defaultTiles);
+  // get random tiles from the bag, only push the letter to the player's tiles
+    this.randomTiles.push(tile[0]);
+  // remove those tiles from the bag so they cannot be drawn again
+  // the index 1 of tile is the random index that cause it to be drawn...
+  // this is the index in defaultTiles
+    this.defaultTiles.splice(tile[1],1);
+  }
+  return this.randomTiles;
+
+};
+
+// helper function to wrap ugly random
+function sample(array) {
+  var index = Math.floor ( Math.random() * array.length );
+  return [array[index], index];
+}
+
+Scrabble.prototype.score = function(word) {
+  this.word = word.toLowerCase(); // lowercase input for comparing
+
+    if (word === "") {
+      return 0;
+    } else {
+      var score = 0;
+      for(var i=0; i<this.word.length; i++) {
+        score += this.letterScore[this.word[i]];
+      }
+
+      if (this.word.length >= this.maxLength) {
+        score += this.bonus;
+      }
+
+    return score;
+    }
+};
+
+Scrabble.prototype.highestScore = function(wordArr) {
+  var highScore = 0;
+  var highScoreWord = "";
+
+  for(var i=0; i<wordArr.length; i++) {
+    var word = wordArr[i];
+    var score = this.score(word);
+
+    if (score > highScore) {
+      highScore = score;
+      highScoreWord = word;
+    } else if (score == highScore) {
+// if the top score is tied between multiple words,
+// pick the one with the fewest letters.
+      if (word.length < highScoreWord.length) {
```

Comment:
It would make more sense to add this to the `else if` above with an `&&` rather than having another `if` inside